File: JavaBotTest/Bot.java
import java.util.Scanner;


public abstract class Bot {
    private Scanner scan = new Scanner(System.in);
    // settings variables
    private int timeBank;
    private int timePerMove;
    private String playerNames;
    private String yourBot;
    private int yourBotId;
    private int fieldWidth;
    private int fieldHeight;

    // game variables
    private int gameRound;
    private String gameField;


    abstract String getAction();

    public void updateSettings(String type, String value) {
        switch (type) {
        case "timebank":
            this.timeBank = Integer.parseInt(value);
            break;
        case "time_per_move":
            this.timePerMove = Integer.parseInt(value);
            break;
        case "player_names":
            this.playerNames = value;
            break;
        case "your_bot":
            this.yourBot = value;
            break;
        case "your_botid":
            this.yourBotId = Integer.parseInt(value);
            break;
        case "field_width":
            this.fieldWidth = Integer.parseInt(value);
            break;
        case "field_height":
            this.fieldHeight = Integer.parseInt(value);
            break;
        }
    }

    public void updateGame(String type, String value) {
        switch (type) {
        case "round":
            this.gameRound = Integer.parseInt(value);
            break;
        case "field":
            this.gameField = value;
        }
    }

    public void runBot() {
        while (scan.hasNextLine()) {
            String line = scan.nextLine();

            if (line.length() == 0) continue;

            String[] command = line.split(" ");
            switch (command[0]) {
            case "settings":
                this.updateSettings(command[1], command[2]);
                break;

            case "update":
                this.updateGame(command[1], command[2]);
                break;

            case "action":
                this.getAction();
                break;
                
            }
        }
    }

    public static void main(String[] args) {
        
    }
}

File: Bots/PythonBots/Bot.py
import sys

class Bot():

    def __init__(self, your_botid):
        self.time_bank = 0
        self.time_per_move = 0
        self.playerNames = ""
        self.your_bot = ""
        self.your_botid = your_botid
        self.field_width = 16
        self.field_height = 16
        self.game_round = 0
        self.game_field = ""
    
    def get_action(self):
        raise NotImplementedError("Must implement get action")

    def on_dead(self):
        raise NotImplementedError("Must implement on dead method")

    def set_botid(self, your_bot_id):
        self.your_bot_id = your_bot_id

    def update_settings(self, setting, value):
        if setting == 'timebank':
            self.time_bank = value            
        elif setting == 'time_per_move':
            self.time_per_move = value
        elif setting == 'player_names':
            self.player_names = value
        elif setting == 'your_bot':
            self.your_bot = value
        elif setting == 'your_botid':
            self.your_botid = value
        elif setting == 'field_width':
            self.field_width = value
        elif setting == 'field_height':
            self.field_height = value            

    def update_game(self, setting, value):
        if setting == 'round':
            self.game_round = value
        elif setting == 'field':
            self.game_field = value
            

    def run_command(self, command):
        command_type = command[0]

        if command_type == 'settings':
            self.update_settings(command[1], command[2])
        elif command_type == 'update':
            self.update_game(command[2], command[3])
        elif command_type == 'action':
            if command[3] == self.your_botid:                
                action = self.get_action()            
                print('%s' % action)
        elif command_type == 'dead':
            if command[2] != self.your_botid:
                print('reset')        
    
    def run_bot(self):
        not_finished = True
        while(not_finished):
            try:
                current_line = sys.stdin.readline().split()                
                self.run_command(current_line)                    
            except EOFError:
                break
            except KeyboardInterrupt:
                raise
            #except:
                # don't raise error or return so that bot attempts to stay alive
                #traceback.print_exc(file=sys.stderr)
                #sys.stderr.flush()

